### PR TITLE
fix build without dune-istl

### LINKED ISSUE #455: dune-istl is required rather than a suggestion
In the dune module file states that this module may optionally use `dune-istl`, however, when it comes to compilation the project fails because there is a bunch of places where dune istl objects are used regartheless if the `dune-istl` was found or not.

### PR DESCRIPTION
**Pull request by akva2**
Closes #455

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: opm/grid/common/GridPartitioning.cpp [hunk 1]
```diff
@@ -242,6 +244,7 @@ namespace Dune
         }
     }
 
+#if HAVE_DUNE_IST
```
Comment (blattms) [inline opm/grid/common/GridPartitioning.cpp]: typo

**Comment by akva2** [comment]
can you just do it like you want it then?

**Comment by blattms** [comment]
Sure, will do.

**Pull request closed without merging**